zerver/views/streams.py

```diff
@@ -325,7 +325,7 @@ def update_stream_backend(
         raise JsonableError(_("A default channel cannot be private."))
 
     # Ensure that a moderation request channel isn't set to public.
-    if not proposed_is_private and user_profile.realm.get_moderation_request_channel() == stream:
+    if not proposed_is_private and user_profile.realm.moderation_request_channel == stream:
```

Comment:
I think we probably also need a commit to prevent deactivating a channel that's used for a realm-level setting that you don't have permission to modify, since this will have the side effect of modifying that realm-level setting.

(This should now be possible with the new per-channel administrator permissions -- add a normal user to that group for the moderation request channel and have them try to deactivate it would be the test)

Reply:
Existing bug, unrelated to this PR.

Filed an [issue on CZO](https://chat.zulip.org/#narrow/channel/9-issues/topic/deactivate.20channel.20permission/near/2075703) in case there's some UX feedback. I'll fix this separately.